File: 4-medianTwoSortedArrays/Go/medianTwoSortedArrays_test.go
package medianTwoSortedArrays

import (
	"testing"

	"github.com/google/go-cmp/cmp"
)

func TestArray(t *testing.T){
	for _, c := range[]struct{
		arr1, arr2 []int
		want float32
	}{
		{[]int{0}, []int{1,2}, 1 },
	}{
		got := medianTwoSortedArrays(c.arr1, c.arr2)
		if !cmp.Equal(got, float64(c.want)) {
			t.Errorf("medianTwoSortedArrays(%v, %v) got == %v, want %v", c.arr1, c.arr2, got, c.want)
		}
	}
}

File: 4-medianTwoSortedArrays/Cpp/medianTwoSortedArrays_test.cpp
#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
#include "doctest.h"

#include "medianTwoSortedArrays.cpp"

TEST_CASE("TESTING"){
    std::vector<std::vector<int>> inArrays = { {0}, {0}, {2,4,3}, {5,6,4}, {8}, {23,-10,-9} };
    std::vector<float> wants = { 0, 2.5, 4, 5.5, -0.5 };
  
    for (auto i = 0; i < wants.size(); i++)
    {
        auto a1 = inArrays[i];
        auto a2 = inArrays[i+1];
        auto want = wants[i];
        auto got = medianTwoSortedArrays(a1,a2);

        CHECK(got == want);
    }
}